lib/src/Pos/Util/Servant.hs

```diff
@@ -699,6 +713,55 @@ instance ReportDecodeError api =>
     reportDecodeError _ msg =
         (ApiNoParamsLogInfo msg, reportDecodeError (Proxy @api) msg)
 
+
+-------------------------------------------------------------------------
+-- Custom query flag
+-------------------------------------------------------------------------
+
+-- This type is used as a helper to implement custom query flags.
+-- Instead of using `QueryFlag "some_flag"` which should serialize
+-- into boolean flag now we can say `CustomQueryFlag "some_flag" SomeFlag`
+-- where SomeFlag has instance of Flaggable. This way we won't be using
+-- Boolean type for all flags but we can implement custom type.
+data CustomQueryFlag (sym :: Symbol) flag
+
+-- TODO (akegalj): add roundtrip test
```

Comment:
Seems that you've already done it, right :D ?

Reply:
ah, right - will remove it